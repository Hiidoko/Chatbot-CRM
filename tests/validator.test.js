const { validateCliente, validateClienteParcial } = require('../validators/clienteValidator');

function baseCliente() {
  return {
    nome: 'João da Silva',
    email: 'joao@example.com',
    telefone: '11999999999',
    cidade: 'São Paulo',
    maquina: 'Máquina A',
    horario: 'Manhã',
    status: 'novo'
  };
}

describe('ClienteValidator (schema unificado)', () => {
  test('valida cliente completo OK', () => {
    const r = validateCliente(baseCliente());
    expect(r.valid).toBe(true);
    expect(r.errors).toHaveLength(0);
  });

  test('detecta email inválido', () => {
    const c = baseCliente();
    c.email = 'invalido@';
    const r = validateCliente(c);
    expect(r.valid).toBe(false);
    expect(r.errors.map(e=>e.field)).toContain('email');
    // Snapshot somente dos erros para reduzir fragilidade
    expect({ errors: r.errors }).toMatchSnapshot();
  });

  test('partial não exige campos ausentes', () => {
    const r = validateClienteParcial({ email: 'novo@mail.com' });
    expect(r.valid).toBe(true);
  });
});
