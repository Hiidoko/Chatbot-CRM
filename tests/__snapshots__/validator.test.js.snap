// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClienteValidator (schema unificado) detecta email inválido 1`] = `
{
  "errors": [
    {
      "field": "email",
      "message": "E-mail inválido.",
    },
  ],
}
`;
